src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -2533,6 +2582,59 @@ string GetBindingOptions()
 			}
 		}
 
+		private void BuildCustomMarkupExtensionPropertyValue(IIndentedStringBuilder writer, XamlMemberDefinition member, string prefix)
+		{
+			Func<string, string> formatLine = format => prefix + format + (prefix.HasValue() ? ";\r\n" : "");
+
+			var propertyValue = GetCustomMarkupExtensionValue(member);
+
+			if (propertyValue.HasValue())
+			{
+				var formatted = formatLine($"{member.Member.Name} = {propertyValue}");
+
+				writer.AppendLine(formatted);
+			}
+		}
+
+		private string GetCustomMarkupExtensionValue(XamlMemberDefinition member)
+		{
+			// Get the type of the custom markup extension
+			var markupType = member
+				.Objects
+				.FirstOrDefault(o => IsCustomMarkupExtensionType(o.Type));
+
+			// Build a string of all its properties
+			var porperties = markupType
+				.Members
+				.Select(m =>
+				{
+					var resourceName = GetStaticResourceName(m);
+
+					var value = resourceName != null
+						? resourceName
+						: BuildLiteralValue(m, owner: member);
+
+					return "{0} = {1}".InvariantCultureFormat(m.Member.Name, value);
+				})
+				.JoinBy(", ");
+
+			// Get the full globalized namespaces for the custom markup extension and also for IMarkupExtensionOverrides
+			var markupTypeFullName = GetGlobalizedTypeName(FindType(markupType.Type).GetFullName());
+			var xamlMarkupFullName = GetGlobalizedTypeName(XamlConstants.Types.IMarkupExtensionOverrides);
+
+			// Get the attribute from the custom markup extension class then get the return type specifed with MarkupExtensionReturnTypeAttribute
+			var attributeData = FindType(markupType.Type).FindAttribute(XamlConstants.Types.MarkupExtensionReturnTypeAttribute);
```

Comment:
Put the result of `FindType(markupType.Type)` in a variable, it is used multiple times

Reply:
It's now a GetType stored in a variable.